src/transforms/deployment.ts

```diff
@@ -26,19 +27,30 @@ function mapState(state: string): string {
 	}
 }
 
-function mapEnvironment(environment: string): string {
-	// We need to map the environment of a GitHub deployment back to a valid deployment environment in Jira.
-	// https://docs.github.com/en/actions/reference/environments
-	// GitHub: does not have pre-defined values and users can name their environments whatever they like. We try to map as much as we can here and log the unmapped ones.
-	// Jira: Can be one of unmapped, development, testing, staging, production
-	const isEnvironment = (envNames) =>
-		envNames.some(envName => environment.localeCompare(envName, undefined, { sensitivity: "base", ignorePunctuation: true }) === 0);
+// We need to map the environment of a GitHub deployment back to a valid deployment environment in Jira.
+// https://docs.github.com/en/actions/reference/environments
+// GitHub: does not have pre-defined values and users can name their environments whatever they like. We try to map as much as we can here and log the unmapped ones.
+// Jira: Can be one of unmapped, development, testing, staging, production
+export function mapEnvironment(environment: string): string {
+	const isEnvironment = (envNames: string[]): boolean => {
+		// Matches any of the input names exactly
+		const exactMatch = envNames.join("|");
+		// Matches separators within environment names, e.g. "-" in "prod-east" or ":" in "test:mary"
+		const separator = "[^a-z0-9]";
+		// Matches an optional prefix, followed by one of the input names, followed by an optional suffix.
+		// This lets us match variants, e.g. "prod-east" and "prod-west" are considered variants of "prod".
+		const envNamesPattern = RegExp(
+			`^(.*${separator})?(${exactMatch})(${separator}.*)?$`,
+			"i"
+		);
+		return envNamesPattern.test(_.deburr(environment));
+	};
```

Comment:
I've split out some helper variables and added comments - hopefully more readable now :)